Lib/test/test_concurrent_futures.py

```diff
@@ -290,6 +290,35 @@ def _assert_logged(self, msg):
 create_executor_tests(FailingInitializerMixin)
 
 
+@unittest.skipIf(sys.platform == "win32", "Resource Tracker doesn't run on Windows")
+class FailingInitializerResourcesTest(unittest.TestCase):
+    """
+    Source: https://github.com/python/cpython/issues/104090
+    """
+
+    def _test(self, test_class):
+        runner = unittest.TextTestRunner()
+        result = runner.run(test_class('test_initializer'))
+
+        self.assertEqual(result.testsRun, 1)
+        self.assertEqual(result.failures, [])
+        self.assertEqual(result.errors, [])
```

Comment:
Uhm, why not do the simple thing and add these checks somewhere in `FailingInitializerMixin`?
It should also avoid running those tests twice (which I assume this does).

Reply:
The result object exists when we run the test using Test Runner, which is not the case on `FailingInitializerMixin`.
This is used for safety, ensuring the inner test passed before we check the resource tracker

Reply:
Yes, but this is needlessly complex (and might interfere with other test options/customizations). Let's do the check at the end of said tests.

Reply:
@pitrou I don't understand how you want those checks to be since we can check it only after a test run with the returned result, which is not the case in regular tests. Anyway, I'm removing those checks since it's not mandatory and just for safety